lib/SIL/Verifier/LinearLifetimeChecker.cpp

```diff
@@ -305,10 +305,21 @@ void State::checkForSameBlockUseAfterFree(Operand *consumingUse,
   // must be instructions in the given block. Make sure that the non consuming
   // user is strictly before the consuming user.
   for (auto *nonConsumingUse : nonConsumingUsesInBlock) {
-    if (std::find_if(consumingUse->getUser()->getIterator(), userBlock->end(),
-                     [&nonConsumingUse](const SILInstruction &i) -> bool {
-                       return nonConsumingUse->getUser() == &i;
-                     }) == userBlock->end()) {
+    if (std::find_if(
+            consumingUse->getUser()->getIterator(), userBlock->end(),
+            [&nonConsumingUse](const SILInstruction &i) -> bool {
+              if (nonConsumingUse->getUser() != &i) {
+                return false;
+              }
+              if (!isa<TermInst>(&i)) {
+                return true;
+              }
```

Comment:
I need to think about this. This is used for more than just the reborrow verifier.